src/Data/Array.purs

```diff
@@ -1289,3 +1293,78 @@ unsafeIndex :: forall a. Partial => Array a -> Int -> a
 unsafeIndex = unsafeIndexImpl
 
 foreign import unsafeIndexImpl :: forall a. Array a -> Int -> a
+
+-- | Takes an arrays of length n and returns an array of Tuples with length n-1.
+-- | It's a "sliding window" view of this array with a window size of 2 and a step size of 1.
```

Comment:
I don't think we should explicitly address the case where someone modifies the array, because the `Array` type is immutable and the language doesn't give you any way of mutating them unless you explicitly opt in to unsafe mechanisms like ST's `unsafeFreeze/unsafeThaw` or the FFI.

Reply:
done

Reply:
> I don't think we should explicitly address the case where someone modifies the array

Yeah, that was likely an overreaction on my part because of the word, "view." The latest rendering looks good to me.

Reply:
No worries - aren't we all afraid of mutable stuff..? :P